libs/data-mapper/src/lib/components/edge/ConnectionEdge.tsx

```diff
@@ -50,19 +51,78 @@ export const ConnectionEdge = (props: EdgeProps) => {
   const { id, sourceX, sourceY, sourcePosition, targetX, targetY, targetPosition, selected } = props;
   const intl = useIntl();
   const styles = useStyles();
+  const reactFlowNodes = useNodes();
 
   const [isHovered, setIsHovered] = useState(false);
 
-  const [edgePath, labelX, labelY] = useMemo(
-    () => getSmoothStepPath({ sourceX: sourceX + 10, sourceY, sourcePosition, targetX: targetX - 10, targetY, targetPosition }),
-    [sourcePosition, sourceX, sourceY, targetX, targetY, targetPosition]
-  );
-
   const insertFnLoc = intl.formatMessage({
     defaultMessage: 'Insert function',
     description: 'Message to insert function',
   });
 
+  const baseEdgeProperties = useMemo(
+    () => ({
+      sourcePosition,
+      targetPosition,
+      sourceX: sourceX + 10,
+      sourceY,
+      targetX: targetX - 10,
+      targetY,
+    }),
+    [sourcePosition, targetPosition, sourceX, sourceY, targetX, targetY]
+  );
+
+  const smartEdge = useMemo(
+    () =>
+      getSmartEdge({
+        ...baseEdgeProperties,
+        nodes: reactFlowNodes,
+        options: {
+          // nodePadding
```

Comment:
Are these comments still important?

Reply:
Yeah, intentionally left them there for quick/easy awareness of their existence as they may be critical and/or volatile settings in the future if any advanced tweaking is done to layouting